src/seq/minhash.jl

```diff
@@ -0,0 +1,118 @@
+# MinHash
+# ========
+#
+# Functions to generate MinHash sketches of biological sequences
+#
+# This file is a part of BioJulia.
+# License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+using DataStructures: SortedSet
+
+"""
+MinHash Sketch type
+
+MinHash sketches are a sorted set of kmer hashes, containing the smallest `s`
+hashes for kmers of a given length. The type contains two parameters:
+
+* .sketch: a sorted set of hashes
+* .kmersize: the length of kmers used to generate the sketch
+"""
+type MinHashSketch
+    sketch::Vector{UInt64}
+    kmersize::Int
+
+    function MinHashSketch(sketch::Vector, kmersize::Int)
+        length(sketch) > 0 || error("Sketch cannot be empty")
+        kmersize > 0 || error("Kmersize must be greater than 0")
+        new(sketch, kmersize)
+    end
+end
+
+Base.getindex(s::MinHashSketch, part) = getindex(s.sketch, part)
+Base.length(s::MinHashSketch) = length(s.sketch)
+Base.size(s::MinHashSketch) = (length(s), s.kmersize)
+Base.start(s::MinHashSketch) = start(s.sketch)
+Base.next(s::MinHashSketch, state) = next(s.sketch, state)
+Base.done(s::MinHashSketch, state) = done(s.sketch, state)
+
+function Base.:(==)(a::MinHashSketch, b::MinHashSketch)
+    a.kmersize == b.kmersize && a.sketch == b.sketch
+end
+
+# A seqence and its reverse complement should be the same, so take the smallest
+# hash of a seq or its reverse complement.
+function revcomphash(kmer::Kmer)
+    k = minimum((kmer, reverse_complement(kmer)))
+    return hash(k)
+end
+
+
+function kmerminhash(seq::BioSequence, kmerset, kmerhashes::Vector{UInt64}, k::Int, s::Int)
```

Comment:
@bicycle1885 Though now I'm going to modify the code, what about the `minhash()` functions that take groups of sequences rather than just one? Will it still be performant if I do something like:

```
function kmerminhash{k, T<:BioSequence}(::Type{DNAKmer{k}}, seqs::Vector{T}, s::Int)
    # generate first `s` kmers
    kmerhashes = UInt64[]
    iter = [each(DNAKmer{k}, seq) for seq in seqs]
    ...
```

?

Or would it make more sense to make a separate iterator function?